.github/workflows/publish.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish Package
+on:
+    push:
+        branches:
+            - main
+        paths:
+            - 'package.json'
+jobs:
+    publish:
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            packages: write
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v2
+
+            - name: Check if version has been updated
+              id: check
+              uses: EndBug/version-check@v1
+              with:
+                  diff-search: true
+                  token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Secret is automatically generated by github https://docs.github.com/en/actions/security-guides/automatic-token-authentication